basics_scenarios/sitewise_scenario/SPECIFICATION.md

```diff
@@ -0,0 +1,336 @@
+SiteWise service. The scenario creates this resource using a CloudFormation template.
+
+## Hello AWS IoT SiteWise
+This program is intended for users not familiar with the AWS IoT SiteWise Service to easily get up and running. The logic is to show use of 
+`listAssetsPaginator`.
+
+## Basics Scenario Program Flow
+The AWS IoT SiteWise Basics scenario executes the following operations.
+
+1. **Create an AWS SiteWise Asset Model**:
+   - Description: This step creates an AWS SiteWise Asset Model by invoking the `createAssetModel` method.
+   - Exception Handling: Check to see if a `ResourceAlreadyExistsException` is thrown. 
+     If it is thrown, get the asset model ID and move on.
+
+2. **Create an AWS IoT SiteWise Asset**:
+   - Description: This operation creates an AWS SiteWise asset.
+   - The method `createAsset` is called to obtain the asset ID.
+   - Exception Handling: Check to see if a `ResourceNotFoundException` is thrown. If so, 
+   display the message and end the program.
+
+3. **Retrieve the property ID values**:
+   - Description: To send data to an asset, we need to get the property ID values for the 
+   the model properties. This scenario uses temperature and humidity properties.
+   - The method `listAssetModelProperties()` is called to retrieve the property ID values.
+   - Exception Handling: Check to see if an `IoTSiteWiseException` is thrown. There are not 
+   many other useful exceptions for this specific call. If so, display the message and end the program.
+
+4. **Send data to an AWS IoT SiteWise Asset**:
+   - Description: This operation sends data to an IoT SiteWise Asset.
+   - This step uses the method `batchPutAssetPropertyValue()`.
+   - Exception Handling: Check to see if a `ResourceNotFoundException` is thrown. If so, display the message and end the program.
+
+5. **Retrieve the value of the IoT SiteWise Asset property**:
+   - Description: This operation gets data from the asset.
+   - This step uses the method `getAssetPropertyValue()`.
+   - Exception Handling: Check to see if a `ResourceNotFoundException` is thrown. If so, display the message and end the program.
+
+6. **Create an IoT SiteWise Portal**:
+   - Description: This operation creates an IoT SiteWise portal.
+   - The method `createPortal` is called.
+   - Exception Handling: Check to see if an `IoTSiteWiseException` is thrown.
+
+7. **Describe the Portal**:
+   - Description: This operation describes the portal and returns a URL for the portal.
+   - The method `describePortal()` is called and returns the URL.
+   - Exception Handling: Check to see if a `ResourceNotFoundException` is thrown. If so, display the message and end the program.
+
+8. **Create an IoTSiteWise Gateway**:
+   - Description: This operation creates an IoT SiteWise Gateway.
+   - The method `createGateway` is called.
+   - Exception Handling: Check to see if an `IoTSiteWiseException` is thrown.
```

Comment:
What to do if the exception is thrown? Move on? or end the program?

Reply:
Fixed